docs/tutorials/openai-text-endpoints.md

```diff
@@ -0,0 +1,113 @@
+<!--
+SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+SPDX-License-Identifier: Apache-2.0
+-->
+
+# Profile OpenAI-Compatible Text APIs Using AIPerf
+
+This guide covers profiling OpenAI-compatible Chat Completions and Completions endpoints with vLLM and AIPerf.
+
+## Start a vLLM server
+
+```bash
+# Pull and run vLLM Docker container:
+docker pull vllm/vllm-openai:latest
+docker run --gpus all -p 8000:8000 vllm/vllm-openai:latest \
+  --model Qwen/Qwen3-0.6B \
+  --reasoning-parser qwen3 \
+  --host 0.0.0.0 --port 8000
+```
+
+```bash
+timeout 900 bash -c 'while [ "$(curl -s -o /dev/null -w "%{http_code}" localhost:8000/v1/chat/completions -H "Content-Type: application/json" -d "{\"model\":\"Qwen/Qwen3-0.6B\",\"messages\":[{\"role\":\"user\",\"content\":\"test\"}],\"max_tokens\":1}")" != "200" ]; do sleep 2; done' || { echo "vLLM not ready after 15min"; exit 1; }
+```
+
+## Profile Chat Completions API
+The Chat Completions API uses the `/v1/chat/completions` endpoint.
+
+### Profile with synthetic inputs
+
+<!-- aiperf-run-vllm-default-openai-endpoint-server -->
+```bash
+aiperf profile \
+    --model Qwen/Qwen3-0.6B \
+    --endpoint-type chat \
+    --endpoint /v1/chat/completions \
+    --streaming \
+    --synthetic-input-tokens-mean 100 \
+    --synthetic-input-tokens-stddev 0 \
+    --output-tokens-mean 200 \
+    --output-tokens-stddev 0 \
+    --url localhost:8000 \
+    --request-count 20
+```
+<!-- /aiperf-run-vllm-default-openai-endpoint-server -->
+
+### Profile with custom input file
+Create a JSONL input file:
+
+<!-- aiperf-run-vllm-default-openai-endpoint-server -->
+
+```bash
+cat <<EOF > inputs.jsonl
+{"texts": ["Hello!", "How are you?"]}
+{"texts": ["Tell me a joke.", "Tell me a story."]}
```

Comment:
kind of a nit, in know the open ai chat completions endpoint has `messages` as a list of dicts, but might it be a bit confusing to show the user as an introduction to the api here the use case where multiple dicts of messages are sent? i would think just one text string per list per line would be less confusing to a beginner?